components/style/properties/properties.mako.rs

```diff
@@ -2665,6 +2663,14 @@ pub fn apply_declarations<'a, F, I>(device: &Device,
                                                  error_reporter);
             % endif
             }
+
+            let writing_mode = get_writing_mode(context.style.get_inheritedbox());
+            context.style.writing_mode = writing_mode;
```

Comment:
Why do you need to move writing mode down here? It doesn't seem to be related, and I suspect whether this is a right thing to do... IIRC font computation somehow depends on writing-mode, although I forget how :/